zcoap.c

```diff
@@ -66,15 +68,21 @@ static int zcoap_request(int sock, u8_t *path, u8_t method, u8_t *payload, u16_t
 			coap_block_transfer_init(&blk_ctx, COAP_BLOCK_1024,
 						BLOCK_WISE_TRANSFER_SIZE_GET);
 		}
-		else if (method == COAP_METHOD_POST || method == COAP_METHOD_PUT) {
-			coap_block_transfer_init(&blk_ctx, COAP_BLOCK_1024, *payload_len);
+		else if ((method == COAP_METHOD_POST || method == COAP_METHOD_PUT) &&
+			  *payload_len > COAP_BLOCK_THRESHOLD) {
+			r = coap_block_transfer_init(&blk_ctx, COAP_BLOCK_1024, *payload_len);
+			if (r != 0) {
+				LOG_ERR("failed to coap_block_transfer_init(%d)\n", r);
+				return 0;
```

Comment:
Its return value may handled by degu_coap_request() case statement, that handles "default". Does it prefer to set COAP_FAILED_TO_RECEIVE_RESPONSE(-1) ?

Reply:
I'll change return value to "code".
COAP_FAILED_TO_RECEIVE_RESPONSE(-1) means "cannot receive response", not system or function error.